### PR TITLE
[Enhancement] Support hrnet frozen stage

### PR DESCRIPTION
**Pull request by sshuair**
## Motivation

Add new feature to support HRNet frozen stage like ResNet.

## Modification

- Add new parameter `frozen_stages` for `HRNet`.
- Add new method `_freeze_stages` for `HRNet`.
- Add new test function `test_hrnet_backbone` to test the frozen_stages.

## BC-breaking (Optional)

No.

## Use cases (Optional)

1. use the frozen_stages from code
    ```python
    model = HRNet(extra, frozen_stages=1) # frozen the first stage
    ```
2. use the frozen_stages from config
    ```python
    _base_ = './fcn_hr18_512x512_20k_voc12aug.py'
    model = dict(
        pretrained='open-mmlab://msra/hrnetv2_w18_small',
        backbone=dict(
            frozen_stages=1,
            extra=dict(
                stage1=dict(num_blocks=(2, )),
                stage2=dict(num_blocks=(2, 2)),
                stage3=dict(num_modules=3, num_blocks=(2, 2, 2)),
                stage4=dict(num_modules=2, num_blocks=(2, 2, 2, 2)))))

    ```

## Checklist

1. Pre-commit used.
2. The modification is covered by complete unit tests.
3. The modification has no potential influence on downstream projects.
4. No need documents.

### CODE REVIEW

**Comment by Junjun2016** [comment]
Hi @sshuair 
Thanks for your nice PR!

**Review by Junjun2016 — APPROVED** [review]
It seems that the unittest of HRNet are missing in our original codebase.
We appreciate it a lot if you could add more unittests for HRNet.

File: mmseg/models/backbones/hrnet.py [hunk 1]
```diff
@@ -575,6 +607,7 @@ def train(self, mode=True):
         """Convert the model into training mode will keeping the normalization
         layer freezed."""
         super(HRNet, self).train(mode)
+        self._freeze_stages()
```
Comment (xvjiarui) [inline mmseg/models/backbones/hrnet.py:610]: Move this line inside `train()`.

Comment (sshuair) [reply mmseg/models/backbones/hrnet.py:610]: @xvjiarui  I did not find any difference between the `resnet` and `hrnet` about `self._freeze_stages()`. Could you give more information?

image

Comment (xvjiarui) [reply mmseg/models/backbones/hrnet.py:610]: Sorry. My bad, I forgot to expand the collapsed code lines.

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**